File: src/2025/CSES/Sorting_searching/stick_lengths.cpp
#include <iostream>
#include <utility>
#include <vector>

using namespace std;

long long find_median(vector<long long> &nums) {
    if (nums.size() % 2 == 1) {
        size_t m = nums.size() / 2;
        return nums[m];
    } else {
        size_t m1 = nums.size() / 2;
        size_t m2 = nums.size() / 2 - 1;
        return (nums[m1] + nums[m2]) / 2;
    }
}

int main() {
    int n;
    cin >> n;

    vector<long long> nums;
    for (int i = 0; i < n; i++) {
        long long num;
        cin >> num;
        nums.push_back(num);
    }

    sort(nums.begin(), nums.end());

    long long med = find_median(nums);
    long long cost = 0;
    for (long long &num : nums) {
        cost += abs(num - med);
    }

    cout << cost << endl;
}

File: src/2025/CSES/src/main/java/org/allan7yin/searching/StickLengths.java
package org.allan7yin.searching;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.Scanner;

public class StickLengths {
    public static long findMedian(List<Long> nums) {
        int size = nums.size();
        if (size % 2 == 1) {
            int mid = size / 2;
            return nums.get(mid);
        } else {
            int mid1 = size / 2;
            int mid2 = size / 2 - 1;
            return (nums.get(mid1) + nums.get(mid2)) / 2;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        List<Long> nums = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            nums.add(scanner.nextLong());
        }

        nums.sort(Comparator.naturalOrder());

        long med = findMedian(nums);
        long cost = 0;
        for (long num : nums) {
            cost += Math.abs(num - med);
        }

        System.out.println(cost);

        scanner.close();
    }
}
